server/models/Capture.js

```diff
@@ -99,8 +108,22 @@ const getCaptures = (captureRepositoryImpl) => (async (filterCriteria = undefine
     return captures.map((row) => { return Capture({...row}); });
 });
 
+
+const applyVerification = (captureRepositoryImpl) => (async (verifyCaptureProcessed) => {

```

Comment:
I am taking  lax approach in consuming messages here...i.e. assuming that messages are processed in the order they are published.  In the event where a approval is followed by a rejection generating two messages, and if they were to be processed out of order, the data will enter a inconsistent state. A robust code would defer out of order message by checking current state with the event's earlier state. 

Given, this specific processing is only meant for milestone1 as a intermediary step in getting data consistent in field-data-service and not as a long term event processing code (this code goes way in milestone2), I am ok with this simple event handling. Please let me know if this concerns you.

Also, it is a reminder that simple boolean values are not robust in expressing state especially when building distributed  services.

Reply:
Thanks for this note.   I see in the documentation that 'Queues in RabbitMQ are FIFO ("first in, first out"). '   So this would presumably disallow events not arrive out of order, but I can nevertheless thing of scenarios with multiple queue consumers that could cause out of order processing.  So let's return to this later.  I would say it's especially important to not create inconsistent state when we start dealing with tokens that move between wallets, and what is shown on the wallet map/app.

Reply:
Agree, if this were a money moving/token trading or some other critical queue, we would rather implement to handle out of order messages. Will return to this again definitely.